cinema/views.py

```diff
@@ -1 +1,70 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import (Genre, Actor, CinemaHall, Movie, MovieSession)
+from cinema.serializers import (GenreSerializer, ActorSerializer,
+                                CinemaHallSerializer, MovieSerializer,
+                                MovieSessionSerializer, MovieListSerializer,
+                                MovieDetailSerializer,
+                                MovieSessionListSerializer,
+                                MovieSessionDetailSerializer)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action in ("list", "retrieve",):
+            queryset = queryset.prefetch_related(
+                "genres").prefetch_related("actors")
+
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+
+        return MovieSerializer
```

Comment:
```suggestion
        return self.serializer_class
```
Ти не даремно його вище визначаєш